pkg/cluster/internal/create/actions/waitforready/waitforready.go

```diff
@@ -97,6 +97,10 @@ func waitForReady(node nodes.Node, until time.Time) bool {
 		if err != nil {
 			return false
 		}
+		// If the server is not ready the answer is empty and this situation must be taken in account
```

Comment:
The API Server *must* be ready by the point this call is used. This is a prerequisite of so many other things.

Reply:
The problem it is not. Or at least even if it is ready I receive sometimes when I run it an empty command Output to the subsequent lines[0] gives an index out of bound exception.

Reply:
Hmm, can you reply to https://github.com/kubernetes-sigs/kind/pull/2578#pullrequestreview-844900787?

I mean, with an unready api-server we also can't do things like install the CNI / storage daemonsets, remove the taint (this one you appear to have added a special wait to), etc.

If the api-server is unhealthy in the environment, that's a problem, but it *shouldn't* be happening once we reach waitForReady here, where we're waiting for node readiness and have long since been relying on api-server readiness. I don't think we're prepared to remove that assumption nor should we need to. The wait before removing the taint might be reasonable but smells like a kubeadm bug that we should probably be fixing upstream https://github.com/kubernetes-sigs/kind/pull/2578#pullrequestreview-844900787